src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Threading;
+
+namespace System.Runtime.InteropServices
+{
+    // ToString() must return debugging string that includes source-name, target name that was loaded, and location details.
+    public class NativeLibrary
+    {
+        public IntPtr Handle { get; set; }
+        public string Name { get; set; } // Will Return the name referenced by the importing assembly
+
+        // Stores a map of assemblies and on-library-load-callbacks assigned to them.
+        // Simulates an additional field of Assembly (which may be added in the future).
+        private static ConditionalWeakTable<Assembly, Func<LoadNativeLibraryArgs, NativeLibrary>> AssemblyToCallbackMap { set; get; }
+
+        public NativeLibrary(string libraryName, IntPtr handle)
```

Comment:
I would like to see some defensiveness in this constructor. If the `libraryName` is null or empty, that is invalid and we should at least assert, but exceptions would be my preference. Same for `handle`.